main/SAPI.c

```diff
@@ -549,17 +549,26 @@ SAPI_API void sapi_initialize_empty_request(void)
 
 static int sapi_extract_response_code(const char *header_line)
 {
-	int code = 200;
+	zend_ulong code = 200;
 	const char *ptr;
 
 	for (ptr = header_line; *ptr; ptr++) {
 		if (*ptr == ' ' && *(ptr + 1) != ' ') {
-			code = atoi(ptr + 1);
+			const char *code_begin = ptr + 1;
+		       	char *code_end;
+
+			code = ZEND_STRTOUL(code_begin, &code_end, 10);
+
+			/* rfc7230 3.1.2 status-code = 3DIGIT */
+			if(*code_begin < '0' || *code_begin > '9' || (code_end - code_begin) != 3) {
```

Comment:
It seems to me that checking the value of `*code_begin` is not really needed, since the previous `if` already ensures that `*code_begin != ' '` and `ZEND_STRTOUL(…, …, 10)` ensures it is a digit; well, could be other whitespace as well. Not really sure, so maybe keeping this check makes sense.

Reply:
I added it because strtoul accepts plus/negative sign in its starting. e.g. strtoul("+15") == 15

Reply:
Ah, right! Very good!